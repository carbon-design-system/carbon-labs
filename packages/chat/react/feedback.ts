/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import Feedback from '../components/feedback/feedback.js';

export const C4AIFeedback = createComponent({
  tagName: 'c4ai-feedback',
  elementClass: Feedback,
  react: React,
  events: {},
});
