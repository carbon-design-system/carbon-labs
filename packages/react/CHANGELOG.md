# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2025-01-30)


### Bug Fixes

* get web-component test for example-button working ([#357](https://github.com/carbon-design-system/carbon-labs/issues/357)) ([5ae7525](https://github.com/carbon-design-system/carbon-labs/commit/5ae7525867850c2e2bc6fd80834b054759437412))


### Features

* script to generate base component in react + web-components ([#354](https://github.com/carbon-design-system/carbon-labs/issues/354)) ([3c24d37](https://github.com/carbon-design-system/carbon-labs/commit/3c24d375f2b8773f7c4dd704197ae6267d919685))
* **storybook:** add root level storybook instance for landing page ([#358](https://github.com/carbon-design-system/carbon-labs/issues/358)) ([473596b](https://github.com/carbon-design-system/carbon-labs/commit/473596b99b5aef9c1dbe261345d24b3033dd56e2))
* **TextHighlighter:** Add TextHighlighter component - React ([#384](https://github.com/carbon-design-system/carbon-labs/issues/384)) ([61dca6b](https://github.com/carbon-design-system/carbon-labs/commit/61dca6ba71aafcab4a101cb21fc898a630c4c333))
