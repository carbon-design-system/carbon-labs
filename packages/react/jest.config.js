/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export default {
  testEnvironment: 'jsdom',
  preset: 'ts-jest',
  moduleNameMapper: {
    '\\.(css|less|scss|sass)$': '<rootDir>/__mocks__/styleMock.js',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  transform: {
    '^.+\\.(ts|tsx)$': 'ts-jest',
    '^.+\\.(js|jsx)$': ['babel-jest', { configFile: './.babelrc' }],
  },
  transformIgnorePatterns: ['/node_modules/(?!@carbon-labs)'],
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],

  // Test file patterns to match
  testMatch: [
    '<rootDir>/src/components/**/__tests__/**/*.[jt]s?(x)',
    '<rootDir>/src/**/?(*.)+(spec|test).[jt]s?(x)',
  ],
};
