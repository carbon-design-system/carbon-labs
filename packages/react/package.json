{
  "name": "@carbon/labs-react",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "description": "React components for Carbon Labs",
  "license": "Apache-2.0",
  "repository": "https://github.com/carbon-design-system/carbon-labs",
  "scripts": {
    "storybook": "storybook dev -p 3000",
    "storybook:build": "storybook build",
    "test": "jest",
    "test:updateSnapshot": "jest -u"
  },
  "files": [
    "es",
    "lib"
  ],
  "peerDependencies": {
    "react": "^16.8.6 || ^17.0.1 || ^18.2.0",
    "react-dom": "^16.8.6 || ^17.0.1 || ^18.2.0"
  },
  "dependencies": {
    "@carbon/react": "^1.71.1",
    "@carbon/styles": "^1.70.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@babel/core": "^7.26.0",
    "@babel/plugin-proposal-export-default-from": "^7.25.9",
    "@babel/plugin-transform-class-properties": "^7.25.9",
    "@babel/plugin-transform-export-namespace-from": "^7.25.9",
    "@babel/plugin-transform-react-constant-elements": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@babel/preset-typescript": "^7.26.0",
    "@mdx-js/react": "^3.1.0",
    "@rollup/plugin-babel": "^6.0.0",
    "@rollup/plugin-commonjs": "^28.0.0",
    "@rollup/plugin-node-resolve": "^16.0.0",
    "@rollup/plugin-typescript": "^12.1.1",
    "@storybook/addon-docs": "^8.4.7",
    "@storybook/addon-essentials": "^8.4.7",
    "@storybook/addon-interactions": "^8.4.7",
    "@storybook/addon-onboarding": "^8.4.7",
    "@storybook/addon-storysource": "^8.4.7",
    "@storybook/addon-webpack5-compiler-babel": "^3.0.3",
    "@storybook/blocks": "^8.4.7",
    "@storybook/react": "^8.4.7",
    "@storybook/react-webpack5": "^8.4.7",
    "@storybook/test": "^8.4.7",
    "@testing-library/jest-dom": "^6.0.0",
    "@testing-library/react": "^16.1.0",
    "@testing-library/user-event": "^14.5.2",
    "@types/babel__core": "^7",
    "@types/babel__preset-env": "^7",
    "@types/jest": "^29.0.0",
    "@types/node": "^16.18.121",
    "@types/prop-types": "^15",
    "@types/react": "^18.3.12",
    "@types/react-dom": "^18.3.1",
    "autoprefixer": "^10.4.20",
    "babel-jest": "^29.7.0",
    "babel-plugin-dev-expression": "^0.2.3",
    "browserslist-config-carbon": "^11.2.0",
    "css-loader": "^7.1.2",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "mini-css-extract-plugin": "^2.9.2",
    "postcss": "^8.4.49",
    "postcss-loader": "^8.1.1",
    "prop-types": "^15.8.1",
    "remark-gfm": "^4.0.0",
    "rollup-plugin-copy": "^3.5.0",
    "rollup-plugin-strip-banner": "^3.0.0",
    "sass-loader": "^16.0.4",
    "storybook": "^8.4.7",
    "storybook-addon-accessibility-checker": "^3.1.61-rc.3",
    "style-loader": "^4.0.0",
    "ts-jest": "^29.2.5",
    "typescript": "^5.2.2",
    "typescript-config-carbon": "^0.3.0",
    "webpack": "^5.97.0"
  }
}
