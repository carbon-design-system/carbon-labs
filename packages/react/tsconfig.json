{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "types": ["jest", "@testing-library/jest-dom"],
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": ["src", "__tests__/**/*"]
}
