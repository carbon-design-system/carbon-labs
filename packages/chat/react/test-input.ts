/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import TestInput from '../components/test-input/test-input.js';

export const C4AITestInput = createComponent({
  tagName: 'c4ai-test-input',
  elementClass: TestInput,
  react: React,
  events: {},
});
