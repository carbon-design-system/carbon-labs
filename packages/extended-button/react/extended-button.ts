/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import ExtendedButton from '../components/extended-button/extended-button.js';

export const C4AIExtendedButton = createComponent({
  tagName: 'c4ai-extended-button',
  elementClass: ExtendedButton,
  react: React,
  events: {},
});
