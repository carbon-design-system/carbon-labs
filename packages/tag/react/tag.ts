/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import Tag from '../components/tag/tag.js';

export const CLABSTag = createComponent({
  tagName: 'clabs-tag',
  elementClass: Tag,
  react: React,
  events: {},
});
