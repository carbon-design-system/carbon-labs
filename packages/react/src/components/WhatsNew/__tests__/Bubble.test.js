/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { Bubble, BubbleHeader } from '../index';

jest.mock('./whats-new.scss', () => ({}));

describe('Bubble', () => {
  const prefix = 'clabs--whats-new';
  beforeEach(() => {
    Element.prototype.scrollIntoView = jest.fn();
  });
  describe('renders as expected - Component API', () => {
    it('should match snapshot', async () => {
      const { container, getByTestId } = render(
        <div>
          <button
            type="button"
            id="TestTargetElement1"
            data-testid="TargetButton1">
            Target button
          </button>
          <Bubble
            data-testid="TestBubble"
            align="bottom"
            target="#TestTargetElement1"
            open={true}>
            <BubbleHeader data-testid="TestBubbleHeader"></BubbleHeader>
          </Bubble>
        </div>
      );

      await waitFor(() => {
        const bubbleElement = getByTestId('TestBubble');
        expect(container).toMatchSnapshot();
        expect(bubbleElement).toBeInTheDocument();
      });
    });

    it('renders children', () => {
      render(
        <div>
          <button
            type="button"
            id="TestTargetElement1"
            data-testid="TargetButton1">
            Target button
          </button>
          <Bubble
            data-testid="TestBubble"
            align="bottom"
            target="#TestTargetElement"
            open={true}>
            <BubbleHeader data-testid="TestBubbleHeader"></BubbleHeader>
            <div>BubbleTestText</div>
          </Bubble>
        </div>
      );
      const childElement = screen.getByText('BubbleTestText');
      expect(childElement).toBeInTheDocument();
    });

    it('applies a custom class', () => {
      const testClassName = 'testClassName9';
      const { getByTestId } = render(
        <div>
          <button
            type="button"
            id="TestTargetElement1"
            data-testid="TargetButton1">
            Target button
          </button>
          <Bubble
            className={testClassName}
            data-testid="TestBubble"
            align="bottom"
            target="#TestTargetElement"
            open={true}></Bubble>
        </div>
      );
      const bubbleElement = getByTestId('TestBubble');
      expect(bubbleElement).toHaveClass(testClassName);
    });

    it('can render without a dropshadow', () => {
      const { getByTestId } = render(
        <div>
          <button
            type="button"
            id="TestTargetElement1"
            data-testid="TargetButton1">
            Target button
          </button>
          <Bubble
            dropShadow={false}
            data-testid="TestBubble"
            align="bottom"
            target="#TestTargetElement"
            open={true}></Bubble>
        </div>
      );
      const bubbleElement = getByTestId('TestBubble');
      expect(bubbleElement).not.toHaveClass(`${prefix}__bubble-drop-shadow`);
    });

    it('can render in high contrast', () => {
      const { getByTestId } = render(
        <div>
          <button
            type="button"
            id="TestTargetElement1"
            data-testid="TargetButton1">
            Target button
          </button>
          <Bubble
            highContrast={true}
            data-testid="TestBubble"
            align="bottom"
            target="#TestTargetElement"
            open={true}></Bubble>
        </div>
      );
      const bubbleElement = getByTestId('TestBubble');
      expect(bubbleElement).toHaveClass(`${prefix}__bubble-high-contrast`);
    });

    it('should not render without a target', () => {
      render(
        <Bubble
          highContrast={true}
          data-testid="TestBubble"
          align="bottom"
          open={true}>
          BubbleTestText
        </Bubble>
      );
      const bubbleElement = screen.queryByText('BubbleTestText');
      expect(bubbleElement).not.toBeInTheDocument();
    });
    it('should not be visible with a malformed target', () => {
      render(
        <Bubble
          target="#DoesntExist"
          highContrast={true}
          data-testid="TestBubble"
          align="bottom"
          open={true}>
          BubbleTestText
        </Bubble>
      );
      const bubbleElement = screen.queryByText('BubbleTestText');
      expect(bubbleElement).toBeInTheDocument();
      expect(bubbleElement).toHaveClass(`${prefix}__bubble-hidden`);
    });
  });
  describe('BubbleHeader - Component API', () => {
    it('renders with additional props', () => {
      const { getByTestId } = render(
        <BubbleHeader data-testid="TestBubbleHeader"></BubbleHeader>
      );
      const bubbleHeaderElement = getByTestId('TestBubbleHeader');
      expect(bubbleHeaderElement).toBeInTheDocument();
    });

    it('applies a custom class', () => {
      const testClassName = 'testClassName44';
      const { getByTestId } = render(
        <BubbleHeader
          className={testClassName}
          data-testid="TestBubbleHeader"></BubbleHeader>
      );

      const bubbleHeaderElement = getByTestId('TestBubbleHeader');
      expect(bubbleHeaderElement).toHaveClass(testClassName);
    });

    it('renders children', () => {
      render(
        <BubbleHeader>
          <div>BubbleTestText</div>
        </BubbleHeader>
      );

      const childElement = screen.getByText('BubbleTestText');
      expect(childElement).toBeInTheDocument();
    });
  });
});
