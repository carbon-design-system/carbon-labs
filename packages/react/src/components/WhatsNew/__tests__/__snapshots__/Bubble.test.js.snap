// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble renders as expected - Component API should match snapshot 1`] = `
<div>
  <div>
    <button
      data-testid="TargetButton1"
      id="TestTargetElement1"
      type="button"
    >
      Target button
    </button>
    <div
      class="cds--white clabs--whats-new__bubble clabs--whats-new__bubble-open clabs--whats-new__bubble-drop-shadow clabs--whats-new__bubble-hidden"
      data-testid="TestBubble"
    >
      <div
        class="clabs--whats-new__bubble__arrow"
      />
      <header
        class="clabs--whats-new__bubble__header"
        data-testid="TestBubbleHeader"
      />
    </div>
  </div>
</div>
`;
